src/event-bus/EventBus.ts

```diff
@@ -0,0 +1,44 @@
+export type Subscriber = (message: Message) => void;
+export interface Message {
+    key?: any;
```

Comment:
I have removed key

Reply:
I agree that `key` does not serve a general purpose and have removed it. 

One drawback of (1) or (2) is that it requires plugins to execute a linear search on `performance.getEntries()` to find a matching hash or tuple. Or to maintain a map. With `key`, they get the matching entry directly.

Reply:
Issue: 

(1) we cannot reliably create a `id` from <name, entryType, startTime, duration> because `recordResourceURL` is configurable. If turned off, we cannot match `PerformanceEntry.name` to `ResourceEvent.targetUrl` because the latter does not exist. If we do (1), it would have to use id=<entryType, startTime, duration>, which is still a decent key. 

(2) is the same thing as dispatching `{key: performanceEntry, payload:...}`, since the object hash is implied in javascript. Manually creating the hash would require [object-hash](https://github.com/puleos/object-hash) (~16M weekly downloads) but has size=~60kb. Doing our own implementation will be smaller but not very performant. 

In my opinion, if the goal is for key(performanceEntry) === key(resourceEvent), then **the most serviceable option is option (1) with key=<entryType, startTime, duration>**